src/internal/middleware.js

```diff
@@ -44,9 +44,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     throw new Error('`options.onerror` passed to the Saga middleware is not a function!')
   }
 
+  if(options.processAction && !is.func(options.processAction)) {
+    throw new Error('`options.processAction` passed to the Saga middleware is not a function!')
+  }
+
   function sagaMiddleware({getState, dispatch}) {
     runSagaDynamically = runSaga
     const sagaEmitter = emitter()
+    const processAction = options.processAction || ((action) => action)
```

Comment:
`identity` is already implemented in `utils` so you can import it from there - https://github.com/yelouafi/redux-saga/blob/250e722419f9fcaade729c51aaf5d97a3de2b16a/src/internal/utils.js#L11